lnwallet/channel.go

```diff
@@ -1864,40 +1864,20 @@ func (lc *LightningChannel) fetchCommitmentView(remoteChain bool,
 	// revocation key, we take the opposite party's revocation base point
 	// and combine that with the current commitment point.
 	if remoteChain {
-		delayKey = TweakPubKey(lc.remoteChanCfg.DelayBasePoint,
-			commitPoint)
-		paymentKey = TweakPubKey(lc.localChanCfg.PaymentBasePoint,
-			commitPoint)
-		revocationKey = DeriveRevocationPubkey(
-			lc.localChanCfg.RevocationBasePoint,
-			commitPoint,
-		)
-
 		delay = uint32(lc.remoteChanCfg.CsvDelay)
 		delayBalance = theirBalance.ToSatoshis()
 		p2wkhBalance = ourBalance.ToSatoshis()
 	} else {
-		delayKey = TweakPubKey(lc.localChanCfg.DelayBasePoint,
-			commitPoint)
-		paymentKey = TweakPubKey(lc.remoteChanCfg.PaymentBasePoint,
-			commitPoint)
-		revocationKey = DeriveRevocationPubkey(
-			lc.remoteChanCfg.RevocationBasePoint,
-			commitPoint,
-		)
-
 		delay = uint32(lc.localChanCfg.CsvDelay)
 		delayBalance = ourBalance.ToSatoshis()
 		p2wkhBalance = theirBalance.ToSatoshis()
 	}
 
-	// TODO(roasbeef); create all keys unconditionally within commitment
-	// store in commitment, will need all when doing HTLC's
-
 	// Generate a new commitment transaction with all the latest
 	// unsettled/un-timed out HTLCs.
-	commitTx, err := CreateCommitTx(lc.fundingTxIn, delayKey, paymentKey,
-		revocationKey, delay, delayBalance, p2wkhBalance, dustLimit)
+	commitTx, err := CreateCommitTx(lc.fundingTxIn, keys.delayKey,
+		keys.paymentKey, keys.revocationKey, delay, delayBalance, p2wkhBalance,
```

Comment:
Might as well just pass in the full `commitKeyRing` here instead of referencing most of it's members as parameters.